### PR TITLE
Don't create a COM weak reference if the object is an aggregated COMWrappers RCW.

### LINKED ISSUE #1025: Taking a WeakReference to a C# object that inherits a native WinRT object crashes the app.
**Describe the bug**
WeakReferences to managed objects inheriting native objects causes InvalidCastExceptions and/or access violations.

**To Reproduce**
1. Open the attached project in VS: [WeakRefRepro.zip](https://github.com/microsoft/CsWinRT/files/7369759/WeakRefRepro.zip)
2. Click "Click Me" repeatedly.

**Expected behavior**
No crash/exception.

**Version Info**
CsWinRT 1.3.5

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This prevents us from trying to re-create a managed object that is a user-defined object deriving from an RCW type from a COM IWeakReference (which fails).

The included test crashes on main with an AV and passes with the changes in this PR.

Fixes https://github.com/microsoft/CsWinRT/issues/1025

### CODE REVIEW

File: src/coreclr/vm/interoplibinterface_comwrappers.cpp [hunk 1]
```diff
@@ -901,6 +904,12 @@ namespace
                             (uniqueInstance
                                 ? ExternalObjectContext::Flags_None
                                 : ExternalObjectContext::Flags_InCache);
+
+                if (flags & CreateObjectFlags::CreateObjectFlags_Aggregated)
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/interoplibinterface_comwrappers.cpp:908]: Can we chain this into the setting of `eocFlags` like `uniqueInstance` above?

File: src/coreclr/vm/interoplibinterface_comwrappers.cpp [hunk 2]
```diff
@@ -1807,6 +1816,7 @@ IUnknown* ComWrappersNative::GetIdentityForObject(_In_ OBJECTREF* objectPROTECTE
     {
         ExternalObjectContext* context = reinterpret_cast<ExternalObjectContext*>(contextMaybe);
         *wrapperId = context->WrapperId;
+        *isAggregated = (context->Flags & ExternalObjectContext::Flags_Aggregated) != 0;
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/interoplibinterface_comwrappers.cpp:1819]:
```suggestion
        *isAggregated = context->IsSet(ExternalObjectContext::Flags_Aggregated);
```

File: src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceNative.cpp [hunk 3]
```diff
@@ -167,8 +167,122 @@ namespace
             return UnknownImpl::DoRelease();
         }
     };
+
+    struct WeakReferenceSource : public IWeakReferenceSource, public IInspectable
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceNative.cpp:171]: Do we need `IInspectable`?

File: src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceTest.cs [hunk 4]
```diff
@@ -28,71 +31,96 @@ public enum WrapperRegistration
...
 
-    public class WeakReferenceableWrapper
+    public unsafe class WeakReferenceableWrapper
     {
         private struct Vtbl
         {
-            public IntPtr QueryInterface;
-            public _AddRef AddRef;
-            public _Release Release;
+            public delegate*<IntPtr, Guid*, IntPtr*, int> QueryInterface;
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceTest.cs:38]: Shouldn't all these functions pointers be `delegate* unmanaged`?

File: src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceNative.cpp (hunk 3 above)
Comment (jkoritzinsky) [reply src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceNative.cpp:171]: Yes we do. We pass this instance to the `WeakReference` struct above, which needs to take an `IInspectable*`.

File: src/coreclr/vm/interoplibinterface_comwrappers.cpp [hunk 5]
```diff
@@ -901,6 +904,12 @@ namespace
                             (uniqueInstance
                                 ? ExternalObjectContext::Flags_None
                                 : ExternalObjectContext::Flags_InCache);
+
+                if (flags & CreateObjectFlags::CreateObjectFlags_Aggregated)
+                {
+                    eocFlags |= ExternalObjectContext::Flags_Aggregated;
+                }
```
Comment (jkoritzinsky) [inline src/coreclr/vm/interoplibinterface_comwrappers.cpp:911]:
```suggestion
                            (uniqueInstance
                                ? ExternalObjectContext::Flags_None
                                : ExternalObjectContext::Flags_InCache) |
                            ((flags & CreateObjectFlags::CreateObjectFlags_Aggregated) != 0
                                ? ExternalObjectContext::Flags_Aggregated
                                : ExternalObjectContext::Flags_None);
```

File: src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceTest.cs [hunk 6]
```diff
@@ -28,71 +31,96 @@ public enum WrapperRegistration
...
+            public delegate*<IntPtr, int> AddRef;
+            public delegate*<IntPtr, int> Release;
         }
```
Comment (jkoritzinsky) [inline src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceTest.cs:41]:
```suggestion
            public delegate* unmanaged<IntPtr, Guid*, IntPtr*, int> QueryInterface;
            public delegate* unmanaged<IntPtr, int> AddRef;
            public delegate* unmanaged<IntPtr, int> Release;
        }
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by AaronRobinsonMSFT** [comment]
@jkoritzinsky Let's start the port to .NET 6 ASAP.

**Pull request merged**